tests/cases/fourslash/tsxGoToDefinitionStatelessFunction2.ts

```diff
@@ -31,10 +31,10 @@
 //// let opt = <[|Main/*sixthTarget*/Button|] wrong />;
 
 verify.goToDefinition({
-    firstTarget: "thirdSource",
-    secondTarget: "thirdSource",
+    firstTarget: "firstSource",
+    secondTarget: "firstSource",
     thirdTarget: "firstSource",
     fourthTarget: "firstSource",
     fifthTarget: "secondSource",
-    sixthTarget: "thirdSource"
+    sixthTarget: "firstSource"

```

Comment:
I'm probably super confused, but I thought normal overload resolution chose the last overload in error cases. Why does this switch to the first overload here?

Reply:
Nope - when overload resolution completely fails we say the declaration is the first signature - see `createUnionOfSignaturesForOverloadFailure`.